coinparam/register.go

```diff
@@ -31,6 +32,26 @@ type Params struct {
 
 	// GenesisHash is the starting block hash.
 	GenesisHash *chainhash.Hash
+	
+	// The function used to calculate the proof of work value for a block
+	PoWFunction func(b []byte) chainhash.Hash
+  
+    // The function used to calculate the difficulty of a given block
+    DiffCalcFunction func(r io.ReadSeeker, height, startheight int32, p *Params) (uint32, error)
+  
+    // The block header to start downloading blocks from
+    StartHeader string
```

Comment:
Can you change this to [80]byte or []byte?  String is easier but could result in runtime errors rather than compile time errors.  Or why not use wire.BlockHeader?  That seems safest